### PR TITLE
Fix: Discrete actions with size > 2 showing only binary values in Godot

### PR DESCRIPTION
**Pull request by Ivan-267**
Discrete actions with size > 2 would display only 0 or 1 even when using the multi discrete spaces.

This fix addresses that by removing the binary conversion (np.greater) in case of using multi discrete spaces. 

A user-friendly error message is also added when trying to mix a discrete action larger than 2 with continuous actions. 

I have quickly tested this on modified "RingPong" example code and it works in cases of:  
1) 1 discrete action with size > 2. 
2) 2 discrete actions with size > 2.
2) Discrete action with size up to 2 + continuous action.

image

However, as I'm not familiar with the code I don't know if this is the right solution, or if it might break something else.

### CODE REVIEW

**Comment by Ivan-267** [comment]
This was deleted by accident, I will have to create a new fork later, I'm still new to Github collaboration. At least the changes are still visible so I will be able to re-implement them easily.

**Comment by edbeeching** [comment]
@Ivan-267 , I am adding you as a contributor to the lib so you can create branches etc directly here without having to fork.

**Comment by Ivan-267** [comment]
As a note, there is more to look into when working with discrete actions than these changes alone. Current sb3 example and e.g. https://github.com/edbeeching/godot_rl_agents/blob/main/docs/ADV_CLEAN_RL.md will produce different values (even for continuous actions which may produce different behaviors, e.g. things like normalization to -1 to 1). I have not yet tried the other libraries/scripts. Exporting to onnx with only discrete actions (size 2 as well) also currently doesn't work (as the logits are exported, but Godot expects a discrete action as a number).

**Comment by visuallization** [comment]
@Ivan-267 Shouldn't we merge this to have at least discrete actions spaces which return more than just 0 and 1? I mean even if it is not working with onnx it seems to improve what we currently have.

**Comment by Ivan-267** [comment]
Is this PR affected by / connected with rllib changes? I haven't tested it much with rllib, but I did more work in the later PR (https://github.com/edbeeching/godot_rl_agents/pull/121).

Specifically this part might be important:

https://github.com/edbeeching/godot_rl_agents/blob/1f718b31f5ac852510dbcba3edfea0ce019c2b0c/godot_rl/core/utils.py#L94-L120

On the commented line I noticed that just sending the direct action as is does not solve the issue in all cases.

In that PR I mentioned some things that still need to be considered (a plugin update is also pending before readding the changes), onnx inference side is affected by changes to this repository. Even with the current implementation, onnx model will behave differently with discrete only actions (causing an action mismatch on export), as the forward method will output the logits during inference, while the sb3 model outputs selected discrete actions on training.

With the linked PR, the onnx model would output discrete actions (e.g. 0,1,2,3) rather than logits, but it would cause an error on the plugin without updating it, as it expects float data type. Rllib would still output logits however with exported onnx.

With the changes to the exported onnx model from sb3, also "true" discrete actions and "mixed discrete actions" (which are continuous actions) need to be handled differently, so I added some changes to the plugin:

https://github.com/edbeeching/godot_rl_agents_plugin/blob/fc67722fc780f10f9ca1b3f259185f4e12b5b423/addons/godot_rl_agents/sync.gd#L130

There might be some other details I would have to recheck my implementation to see.
